src/lib.rs

```diff
@@ -476,28 +543,38 @@ pub struct Category<BlockNumber, Moment, AccountId> {
     /// Whether category is archived.
     archived: bool,
 
-    /*
-     * These next three numbers are not maintained currently, 
-     * they inadverdently made it into the spec without being
-     * part of the user storeis
-     
-    /// Number direct subcategories.
-    num_direct_subcategories: u64,
-
-    /// Number of unmoderated threads directly in this category.
-    num_direct_unmoderated_threads: u64,
-
-    /// Number of moderated threads directly in this category.
-    num_direct_moderated_threads: u64,
-    */
+    /// Number of subcategories (deleted, archived or neither),
+    /// unmoderated threads and moderated threads, _directly_ in this category.
+    /// 
+    /// As noted, the first is unaffected by any change in state of direct subcategory.
+    ///  
+    /// The sum of the latter two only increases, and former is incremented
+    /// for each new thread added to this category. A new thread is added 
+    /// with a `thread_nr_in_category` equal to this sum.
+    /// 
+    /// When there is a moderation
+    /// of a thread, the variables are incremented and decremented, respectively.
+    /// 
+    /// These values are vital for light clients, in order to validate that they are
+    /// not being censored from subcategories or threads in a category.
+    num_direct_subcategories: u32,
+    num_direct_unmoderated_threads: u32,
+    num_direct_moderated_threads: u32,
 
-    /// Parent category, if present, otherwise this category is a root categoryl
-    parent_id: Option<CategoryId>,
+    /// Position as child in parent, if present, otherwise this category is a root category
+    slot_in_parent_category: Option<ChildPositionInParentCategory>,
```

Comment:
The structure has a word `Position` but a field has a word `slot`. Looks inconsistent.

Reply:
Yes, changed variable name.